src/state/middlewares/system/controller.js

```diff
@@ -11,45 +20,76 @@ import {
 	OPERATOR_JOIN,
 	REMOVE_CHAT,
 	CLOSE_CHAT
-} from '../../action-types'
-import { operatorReceiveTyping } from '../../operator/actions'
+} from '../../action-types';
+import { operatorReceiveTyping, sendOperatorChatLog } from '../../operator/actions';
 import {
 	agentReceiveMessage,
 	customerReceiveMessage,
 	customerReceiveTyping,
 	operatorReceiveMessage,
 	receiveMessage,
-} from '../../chatlist/actions'
+	sendCustomerChatLog
+} from '../../chatlist/actions';
 
-const debug = require( 'debug' )( 'happychat-debug:controller' )
+const debug = require( 'debug' )( 'happychat-debug:controller' );
 
-const DEFAULT_MAX_MESSAGES = 100
+const DEFAULT_MAX_MESSAGES = 100;
 
+/**
+ * Chat log caching layer for recently received chat messages.
+ */
 export class ChatLog {
 
+	/**
+	 * @constructor
+	 *
+	 * @param { Object } options - settings for the chat log
+	 * @param { number } options.maxMessages - number of messages to keep in the log
+	 */
 	constructor( options = { maxMessages: DEFAULT_MAX_MESSAGES } ) {
-		this.maxMessages = options.maxMessages
-		this.chats = {}
+		this.maxMessages = options.maxMessages;
+		this.chats = {};
 	}
 
+	/**
+	 * Appends message to chat of `id`
+	 *
+	 * @param { String } id - chat id related tot message
+	 * @param { Object } message - message to append to chat log
+	 * @returns { Promise } resolves whet append is completed
+	 */
 	append( id, message ) {
 		this.chats = assoc( id, compose(
-			slice( - this.maxMessages, Infinity ),
+			slice( -this.maxMessages, Infinity ),
 			append( message )
-		)( this.findLog( id ) ), this.chats )
+		)( this.getLog( id ) ), this.chats );
+		return Promise.resolve();
 	}
 
+	getLog( id ) {
```

Comment:
The difference between `getLog` and `findLog` isn't very obvious from their names... Any way we could make this clearer?